File: lua/character_counts.lua
s = "\u{1f1f7}\u{1f1f1}"
assert(#s == 8)                   -- counts bytes
assert(utf8.len(s) == 2)          -- counts code points

print(utf8.codepoint(s, 1, #s))   -- print each code point
print(string.byte(s, 1, #s))      -- print each byte

File: js/character_counts.js
import assert from "assert/strict"

// Grapheme = flag: Lebanon. 2 characters, 8 bytes
const string = "\u{1f1f1}\u{1f1e7}"
const bytes = Buffer.from(string, "utf8")
const codePoints = [...string]

assert(string.length == 4) // internal rep is 4 chunks of UTF-16)
assert(bytes.length == 8) // ok
assert(codePoints.length == 2) // better
